tests/specification_tests.rs

```diff
@@ -89,14 +90,22 @@ fn run_test_case(test_case: &TestData) -> Result<(), Box<dyn Error>> {
 }
 
 fn run_test_case_serialization_only(test_case: &TestData) -> Result<(), Box<dyn Error>> {
-    let expected_field_value = build_expected_field_value(test_case)?;
-    let actual_result = expected_field_value.serialize();
+    let expected_field_value = build_expected_field_value(test_case);
 
+    // TODO: must_fail has to always fail on creation
+    // As not all cases are moved yet, we take a two-step approach here:
+    // Either creation fails or serialization fails
```

Comment:
If we're just testing serialization, I think we can still create invalid bareItems if we simply don't use try_into, right?

Reply:
If I try to use e.g. the `Integer` value struct in `build_bare_item`, I get

> cannot initialize a tuple struct which contains private fields

Which is the correct way for external usage. Not sure whether we can make the tests "part of the crate module tree" to allow it access to the `pub(crate)` value?

Reply:
Validating keys can be done similar to this PR with a new `Key` value struct that is used with `IndexMap`.

I have drafted the change here, but would probably take that in its own PR: https://github.com/zcei/sfv/compare/validate-bare-items...zcei:sfv:validate-keys

One ergonomic advantage, but not sure whether it justifies the implementation overhead, would be to allow `.get` with any unvalidated `&str` and return `None` early if it can't validate. Only for valid keys we would check back the IndexMap whether it contains the value.